### PR TITLE
Request node settings

### PR DESCRIPTION
**Pull request by miiu96**
- Implemented a mechanism that will request the node settings then the indexer binary will start.

### CODE REVIEW

File: cmd/elasticindexer/main.go [hunk 1]
```diff
@@ -104,6 +108,19 @@ func startIndexer(ctx *cli.Context) error {
 	return nil
 }
 
+func requestSettings(host wsindexer.WSClient, retryDurationInSec uint32) {
+	for {
+		err := host.Send([]byte{}, outport.TopicSettings)
+		if err != nil {
+			time.Sleep(time.Duration(retryDurationInSec) * time.Second)
+
+			log.Debug("request settings", "error", err)
```
Comment (bogdan-rosianu) [inline cmd/elasticindexer/main.go:117]: might print before the `time.Sleep` and also display a `will retry`: 
- `log.Debug("request settings", "error", err)`
=>
- `log.Debug("unable to request settings - will retry", "error", err)`

File: process/dataindexer/dataIndexer.go [hunk 2]
```diff
@@ -189,6 +189,18 @@ func (di *dataIndexer) GetMarshaller() marshal.Marshalizer {
 	return di.headerMarshaller
 }
 
+// RegisterHandlerForSettingsRequest will do nothing
+func (di *dataIndexer) RegisterHandlerForSettingsRequest(_ func()) error {
+	return nil
+}
+
+// CurrentSettings will set the provided settings
+func (di *dataIndexer) CurrentSettings(cfg outport.OutportConfig) error {
```
Comment (bogdan-rosianu) [inline process/dataindexer/dataIndexer.go:198]: rename to `SetCurrentSettings`

Comment (miiu96) [reply process/dataindexer/dataIndexer.go:198]: done.

File: cmd/elasticindexer/main.go (hunk 1 above)
Comment (miiu96) [reply cmd/elasticindexer/main.go:117]: done.

File: cmd/elasticindexer/main.go [hunk 3]
```diff
@@ -104,6 +108,19 @@ func startIndexer(ctx *cli.Context) error {
 	return nil
 }
 
+func requestSettings(host wsindexer.WSClient, retryDurationInSec uint32) {
+	for {
+		err := host.Send([]byte{}, outport.TopicSettings)
+		if err != nil {
+			time.Sleep(time.Duration(retryDurationInSec) * time.Second)
```
Comment (iulianpascalau) [inline cmd/elasticindexer/main.go:115]: please refactor this to allow the normal closing of the node if the settings were not received in time.

File: cmd/elasticindexer/main.go [hunk 4]
```diff
@@ -108,16 +110,23 @@ func startIndexer(ctx *cli.Context) error {
...
-		if err != nil {
-			time.Sleep(time.Duration(retryDurationInSec) * time.Second)
+func requestSettings(host wsindexer.WSClient, retryDurationInSec uint32, close chan os.Signal) bool {
+	timer := time.NewTimer(0)
+	defer timer.Stop()
 
+	for {
+		select {
+		case <-timer.C:
+			err := host.Send([]byte{}, outport.TopicSettings)
```
Comment (iulianpascalau) [inline cmd/elasticindexer/main.go:120]: make([]byte, 0) instead of literal

Comment (miiu96) [reply cmd/elasticindexer/main.go]: done

**Review by bogdan-rosianu — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**